tests/integration/dynamic/nodejs-pulumi-config/index.ts

```diff
@@ -10,9 +10,9 @@ class CustomResource extends pulumi.dynamic.Resource {
 
 class DummyResourceProvider implements pulumi.dynamic.ResourceProvider {
   async create (props: any): Promise<pulumi.dynamic.CreateResult> {
-    var config = JSON.parse(process.env.PULUMI_CONFIG)
+    const config = new pulumi.Config()
```

Comment:
This works currently, do we want to allow this or do we want to prohibit this?

Reply:
Testing this some more, and I think this won't work for Python in the case where a dynamic provider is in `__main__`, because dill will try to pickle `pulumi.Config`, which ends up pulling in `output.py`, which itself depends on grpcio, and that fails pickling because of the protobufs.